src/main/java/gregtech/integration/jei/basic/GTFluidVeinCategory.java

```diff
@@ -0,0 +1,180 @@
+    private String veinName;
+    private int weight;
+    private int[] yields; // the [minimum, maximum) yields
+    private int depletionAmount; // amount of fluid the vein gets drained by
+    private int depletionChance; // the chance [0, 100] that the vein will deplete by 1
+    private int depletedYield; // yield after the vein is depleted
+    private final int SLOT_CENTER = 79;
+    protected final int FONT_HEIGHT = Minecraft.getMinecraft().fontRenderer.FONT_HEIGHT;
+    protected final Map<Integer, String> namedDimensions = WorldGenRegistry.getNamedDimensions();
+    private Supplier<List<Integer>> dimension;
+    private final int textStartX= 5;
+    private int weightLength;
+    private int minYieldLength;
+    private int maxYieldLength;
+    private int depletionChanceLength;
+    private int depletionAmountLength;
+    private int depletedYieldLength;
+    private final int startPosY = 40;
+
+
+    public GTFluidVeinCategory(IGuiHelper guiHelper) {
+        super("fluid_spawn_location",
+                "fluid.spawnlocation.name",
+                guiHelper.createBlankDrawable(176, 166),
+                guiHelper);
+
+        this.slot = guiHelper.drawableBuilder(GuiTextures.SLOT.imageLocation, 0, 0, 18, 18).setTextureSize(18, 18).build();
+
+    }
+
+    @Override
+    public void setRecipe(@Nonnull IRecipeLayout recipeLayout, GTFluidVeinInfo gtFluidVeinInfo, @Nonnull IIngredients ingredients) {
+
+        IGuiFluidStackGroup fluidStackGroup = recipeLayout.getFluidStacks();
+
+        fluidStackGroup.init(0, true, SLOT_CENTER, 19, 16, 16, 1, false, null);
+
+        fluidStackGroup.addTooltipCallback(gtFluidVeinInfo::addTooltip);
+        fluidStackGroup.set(ingredients);
+
+        this.definition = gtFluidVeinInfo.getDefinition();
+        this.veinName = gtFluidVeinInfo.getName();
+        this.weight = gtFluidVeinInfo.getWeight();
+        this.yields = gtFluidVeinInfo.getYields();
+        this.depletionAmount = gtFluidVeinInfo.getDepletionAmount();
+        this.depletionChance = gtFluidVeinInfo.getDepletionChance();
+        this.depletedYield = gtFluidVeinInfo.getDepletedYield();
+
+        this.dimension = GTUtility.getAllRegisteredDimensions(definition.getDimensionFilter());
+
+        //Slight cleanup of the list if Advanced Rocketry is installed
+        if (Loader.isModLoaded(MODID_AR)) {
+            try {
+                int[] spaceDims = DimensionManager.getDimensions(DimensionType.byName("space"));
+
+                //Remove Space from the dimension list
+                for (int spaceDim : spaceDims) {
+                    if (this.dimension.get().contains(spaceDim)) {
+                        this.dimension.get().remove((Integer) spaceDim);
+                    }
+                }
+            } catch (IllegalArgumentException e) {
+                GTLog.logger.error("Something went wrong with AR JEI integration, No DimensionType found");
+                GTLog.logger.error(e);
+            }
+        }
```

Comment:
This part could be moved near `getAllRegisteredDimensions()` and have like **all** version and **processed** version.
Also there is exactly same block of code in `GTOreCategory`.